apps/prairielearn/src/components/QuestionsTable.html.ts

```diff
@@ -241,3 +250,89 @@ export function QuestionsTable({
     </div>
   `;
 }
+
+function CreateQuestionModal({
+  csrfToken,
+  templateQuestions,
+}: {
+  csrfToken: string;
+  templateQuestions: { qid: string; title: string }[];
+}) {
+  return Modal({
+    id: 'createQuestionModal',
+    title: 'Create question',
+    formMethod: 'POST',
+    body: html`
+      <div class="form-group">
+        <label for="title">Title</label>
+        <input
+          type="text"
+          class="form-control"
+          id="title"
+          name="title"
+          required
+          aria-describedby="title_help"
+        />
+        <small id="title_help" class="form-text text-muted">
+          The full name of the question, visible to users.
+        </small>
+      </div>
+      <div class="form-group">
+        <label for="qid">Question identifier (QID)</label>
+        <input
+          type="text"
+          class="form-control"
+          id="qid"
+          name="qid"
+          required
+          aria-describedby="qid_help"
+        />
+        <small id="qid_help" class="form-text text-muted">
+          A short unique identifier for this question, such as "add-vectors" or "find-derivative".
+          Use only letters, numbers, dashes, and underscores, with no spaces.
```

Comment:
Nit: as with other recent PRs, consider adding a `pattern` to enforce this. We can reconsider as part of #9908, but for now, let's keep things consistent.